docs/developer/non-admin-controller.md

```diff
@@ -0,0 +1,48 @@
+# OADP NonAdminBackup Controller
+
+The OADP NonAdminBackup Kubernetes controller manages the lifecycle of custom resources called `NonAdminBackup`. Let's break down the code and understand its functionality.
+
+## Purpose
+
+The primary goal of this controller is to allow users with limited permissions (non-admins) to trigger and manage Velero backups within their namespaces. It acts as an intermediary between the non-admin user and Velero, ensuring that backups are created and managed securely within the defined boundaries.
+
+## Key Components
+
+### NonAdminBackup Custom Resource
+
+* Represents a backup request initiated by a non-admin user.
+* Contains a `BackupSpec` field, which mirrors the Velero `BackupSpec` and defines what to back up.
```

Comment:
With the current discussion it doesn't mirror the Velero `BackupSpec` but it's used to create `BackupSpec`